2-Draft-Accepted/RFCNNN-Importing-Windows-PowerShell-modules-in-PowerShell-Core.md

```diff
@@ -0,0 +1,106 @@
+---
+RFC:
+Author: Andrew Menagarishvili
+Status: Draft
+SupercededBy:
+Version: 0.1
+Area: Language
+Comments Due: 10/10/2019
+Plan to implement: Yes
+---
+
+# Importing Windows PowerShell modules in PowerShell Core
+
+One of the big factors preventing existing Windows PowerShell users from moving to PowerShell Core has been cmdlet coverage. Specifically being able to use existing Windows PowerShell modules in PowerShell Core. Non-compatibility is due to the different .NET runtimes used between Windows PowerShell and PowerShell Core. Ease of UX is important when addressing this problem - working with 'WindowsPS-only' modules from PS Core should not be any different than working with 'PS Core - compatible' modules. Also this should not require any setup from the user.
+
+## Motivation
+
+    As a PowerShell user,
+    I can use modules written for Windows PowerShell in PowerShell Core,
+    so that I could migrate from Windows PowerShell to PowerShell Core.
+
+## User Experience
+
+Example that shows using commands from 'WindowsPS-only' module located in `System32` module path in PS Core:
+```PowerShell
+PS C:\> $PSVersionTable.PSEdition
+Core
+PS C:\> Import-Module DesktopOnlyModuleOnSystem32ModulePath
+VERBOSE: Please note that module 'DesktopOnlyModuleOnSystem32ModulePath' is imported in a remote Windows PowerShell session, so all objects that are returned by commands from this module are deserialized and are not 'live' objects.

```

Comment:
I am not sure that user scripts will works well with deserialized objects. I always had problems when I switched to deserialized objects. I'd expect that we use the technology only for modules compatible with serialization.

Reply:
Unfortunately, I think it's impossible for us to know whether that's the case. I still think we're going to end up with more successful script executions if we automatically load system32 modules (which are *known* to be incompatible post-Server 2019 and are demarcated as such) than if we simply fail to load them